staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -175,22 +181,69 @@ func (q *Type) Done(item interface{}) {
 	q.metrics.done(item)
 
 	q.processing.delete(item)
+	if q.processing.len() == 0 {
```

Comment:
I feel like this could be in an else clause on the following `if` statement?

Reply:
It could. The following:
```go
if q.dirty.has(item) {
        q.queue = append(q.queue, item)
	q.cond.Signal()
} else if q.processing.len() == 0 {
        q.cond.Signal()
}
```
would lead to less `Signal` invocations than the current version